samza-core/src/main/java/org/apache/samza/clustermanager/ContainerManager.java

```diff
@@ -391,6 +402,28 @@ public void registerContainerPlacementAction(ContainerPlacementRequestMessage re
       return;
     }
 
+    /*
+     * When destination host is {@code FORCE_RESTART_LAST_SEEN} its treated as eqvivalent to kill -9 operation for the container
+     * In this scenario container is stopped first and we fallback to normal restart path
+     */
+    if (destinationHost.equals(FORCE_RESTART_LAST_SEEN)) {
+      LOG.info("Issuing a force restart for Processor ID: {} for ContainerPlacement action request {}", processorId, requestMessage);
+      clusterResourceManager.stopStreamProcessor(samzaApplicationState.runningProcessors.get(processorId));
+      writeContainerPlacementResponseMessage(requestMessage, ContainerPlacementMessage.StatusCode.SUCCEEDED,
+          "Successfully issued a stop container request falling back to normal restart path");
+      return;
+    }
+
+    /**
+     * When destination host is {@code LAST_SEEN} its treated as a restart request on the host where container is running
+     * on or has been seen last
+     */
```

Comment:
This comment has the same meaning as the one above 
"if (destinationHost.equals(FORCE_RESTART_LAST_SEEN))" 

Could we write the difference

Reply:
sure